src/content/1.1/code/agda/snippet04.agda

```diff
@@ -1,4 +1,10 @@
-f : A -> B
-g : B -> C
-h : C -> D
-h ∘ (g ∘ f) == (h ∘ g) ∘ f == h ∘ g ∘ f
\ No newline at end of file
+open import Function using (_∘_)
+open import Relation.Binary.PropositionalEquality using (_≡_)
+
+_ :  {A B C D : Set}
+     (f : A → B)
+     (g : B → C)
+     (h : C → D)
+  →  (h ∘ g) ∘ f ≡ h ∘ (g ∘ f)
+
+_ = λ f g h → _≡_.refl
```

Comment:
I think, where possible and where it's not too inconvenient and doesn't require too much extra code, we should try to write snippets that type-check successfully and, where appropriate, include proofs of stated properties (e.g., associativity in this example).  In Haskell and other languages, it's not clear that including formal proofs is a reasonable goal, but in Agda it certainly is. :)

Reply:
Agree on this. OTOH in many places at the beginning they just show syntax - I will re-read this.